prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -497,6 +497,84 @@ class ParserSpec extends AnyFunSpec with Matchers {
+      "sum(http_requests_total{method=\"GET\"} @ 1609746000 offset 5m)" ->
+        "Aggregate(Sum,PeriodicSeries(RawSeries(IntervalSelector(1609746000000,1609746000000),List(ColumnFilter(method,Equals(GET)), ColumnFilter(__name__,Equals(http_requests_total))),List(),Some(300000),Some(300000)),1609746000000,1000000,1609746000000,Some(300000)),List(),List(),List())",
+      "sum(http_requests_total{method=\"GET\"} offset 5m @ 1609746000)" ->
+        "Aggregate(Sum,PeriodicSeries(RawSeries(IntervalSelector(1609746000000,1609746000000),List(ColumnFilter(method,Equals(GET)), ColumnFilter(__name__,Equals(http_requests_total))),List(),Some(300000),Some(300000)),1609746000000,1000000,1609746000000,Some(300000)),List(),List(),List())",
+      "rate(http_requests_total[5m] @ 1609746000)" ->
+       "PeriodicSeriesWithWindowing(RawSeries(IntervalSelector(1609746000000,1609746000000),List(ColumnFilter(__name__,Equals(http_requests_total))),List(),Some(300000),None),1524855988000,1000000,1524855988000,300000,Rate,false,List(),None,List(ColumnFilter(__name__,Equals(http_requests_total))))",
+      "rate(http_requests_total[5m] @ 1609746000 offset 1m)" ->
+        "PeriodicSeriesWithWindowing(RawSeries(IntervalSelector(1609746000000,1609746000000),List(ColumnFilter(__name__,Equals(http_requests_total))),List(),Some(300000),Some(60000)),1524855988000,1000000,1524855988000,300000,Rate,false,List(),Some(60000),List(ColumnFilter(__name__,Equals(http_requests_total))))",
+      "rate(http_requests_total[5m] offset 1m @ 1609746000)" ->
+        "PeriodicSeriesWithWindowing(RawSeries(IntervalSelector(1609746000000,1609746000000),List(ColumnFilter(__name__,Equals(http_requests_total))),List(),Some(300000),Some(60000)),1524855988000,1000000,1524855988000,300000,Rate,false,List(),Some(60000),List(ColumnFilter(__name__,Equals(http_requests_total))))",
+      "rate(foo[10m:30s] @ 12345)" ->
+        "SubqueryWithWindowing(PeriodicSeries(RawSeries(IntervalSelector(1524855390000,1524855960000),List(ColumnFilter(__name__,Equals(foo))),List(),Some(300000),None),1524855390000,30000,1524855960000,None),1524855988000,0,1524855988000,Rate,List(),600000,30000,None)",
+      "rate(foo[10m:30s] offset 1m @ 12345)" ->
+        "SubqueryWithWindowing(PeriodicSeries(RawSeries(IntervalSelector(1524855330000,1524855900000),List(ColumnFilter(__name__,Equals(foo))),List(),Some(300000),None),1524855330000,30000,1524855900000,None),1524855988000,0,1524855988000,Rate,List(),600000,30000,Some(60000))",
+      "rate(foo[10m:30s] @ 12345 offset 1m)" ->
+        "SubqueryWithWindowing(PeriodicSeries(RawSeries(IntervalSelector(1524855330000,1524855900000),List(ColumnFilter(__name__,Equals(foo))),List(),Some(300000),None),1524855330000,30000,1524855900000,None),1524855988000,0,1524855988000,Rate,List(),600000,30000,Some(60000))"
```

Comment:
Timestamps for `TopLevelSubquery` tests are more-or-less expected (given that `getStartForFastSubquery` will shift them around a bit), but these look really wonky. This is expected, though, right?

Reply:
check prometheus